src/TRestTrackEvent.cxx

```diff
@@ -337,6 +405,49 @@ void TRestTrackEvent::GetMaxTrackBoundaries(TVector3& orig, TVector3& end) {
     end = TVector3(endX.X(), endY.Y(), endZ);
 }
 
+///////////////////////////////////////////////
+/// \brief Function calculate the relative Z of the maximum
+/// track to cross check if the track is upwards or downwards
+///
+Double_t TRestTrackEvent::GetMaxTrackRelativeZ() {
+    TRestTrack* tckX = GetMaxEnergyTrackInX();
+    TRestTrack* tckY = GetMaxEnergyTrackInY();
+
```

Comment:
It could happen that `tckX` or `tckY` is null. Obviously not with alphas, but it may happen a weirdo event in between.